datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -62,6 +63,7 @@ pub(crate) enum ExecutionState {
     /// When producing output, the remaining rows to output are stored
     /// here and are sliced off as needed in batch_size chunks
     ProducingOutput(RecordBatch),
+    ProducingBlocks(Option<usize>),
```

Comment:
It would be great to add a comment to explain how other states are transitioning to/from `ProducingBlcoks`
A sketchy one with simple examples would be enough, and we can polish it later